### PR TITLE
(dsl): Support Nested query

### PR DESCRIPTION
**Pull request by mvelimir**
Resolves `nestedQuery` in #91

### CODE REVIEW

**Review by drmarjanovic — DISMISSED** [review]

File: modules/library/src/main/scala/zio/elasticsearch/ElasticQuery.scala [hunk 1]
```diff
@@ -163,6 +184,21 @@ object ElasticQuery {
...
+    query: ElasticQuery[_],
+    scoreMode: Option[ScoreMode],
+    ignoreUnmapped: Option[Boolean]
+  ) extends ElasticQuery[Nested] {
+    def paramsToJson: Json = Obj(
+      "nested" -> Obj(
+        Chunk("path" -> Str(path), "query" -> query.paramsToJson) ++ scoreMode.map(scoreMode =>
+          "score_mode" -> Str(scoreMode.toString.toLowerCase)
+        ) ++ ignoreUnmapped.map("ignore_unmapped" -> Json.Bool(_))
+      )
```
Comment (dbulaja98) [inline modules/library/src/main/scala/zio/elasticsearch/ElasticQuery.scala:198]: Can `Chunk` be replaced with `List`?

File: modules/library/src/main/scala/zio/elasticsearch/ElasticQuery.scala [hunk 2]
```diff
@@ -70,97 +84,129 @@ object ElasticQuery {
     def toJson(implicit EP: ElasticPrimitive[A]): Json = EP.toJson(value)
   }
 
-  def boolQuery: BoolQuery = BoolQuery.empty
-
-  def contains(field: String, value: String): ElasticQuery[Wildcard] =
+  def contains(field: String, value: String): ElasticQuery[Any, Wildcard] =
```
Comment (drmarjanovic) [inline modules/library/src/main/scala/zio/elasticsearch/ElasticQuery.scala:87]: Why do we miss `contains[S](field: Field[S, _])`?

File: modules/library/src/main/scala/zio/elasticsearch/ElasticQuery.scala [hunk 3]
```diff
@@ -70,97 +84,129 @@ object ElasticQuery {
...
+  def matches[A: ElasticPrimitive](field: String, value: A): ElasticQuery[Any, Match] =
     MatchQuery(field, value)
 
-  def range[A](
-    field: Field[_, A],
+  def must[S](queries: ElasticQuery[S, _]*): BoolQuery[S] =
+    BoolQuery[S](filter = Nil, must = queries.toList, should = Nil, boost = None)
+
+  def nested[S, A](path: Field[S, Seq[A]], query: ElasticQuery[A, _]): ElasticQuery[S, Nested] =
+    NestedQuery(path.toString, query, None, None)
```
Comment (drmarjanovic) [inline modules/library/src/main/scala/zio/elasticsearch/ElasticQuery.scala:113]: Let's name params here.

File: modules/library/src/main/scala/zio/elasticsearch/ElasticQuery.scala [hunk 4]
```diff
@@ -70,97 +84,129 @@ object ElasticQuery {
...
 
-  def matches[A: ElasticPrimitive](
-    field: Field[_, A],
+  def matches[S, A: ElasticPrimitive](
+    field: Field[S, A],
     multiField: Option[String] = None,
     value: A
-  ): ElasticQuery[Match] =
+  ): ElasticQuery[S, Match] =
     MatchQuery(field.toString ++ multiField.map("." ++ _).getOrElse(""), value)
```
Comment (drmarjanovic) [inline modules/library/src/main/scala/zio/elasticsearch/ElasticQuery.scala:104]: Let's name params here as well.

File: modules/library/src/main/scala/zio/elasticsearch/ElasticQuery.scala [hunk 5]
```diff
@@ -70,97 +84,129 @@ object ElasticQuery {
...
+
+  def nested(path: String, query: ElasticQuery[_, _]): ElasticQuery[Any, Nested] =
+    NestedQuery(path, query, None, None)
+
+  def range[S, A](
+    field: Field[S, A],
     multiField: Option[String] = None
-  ): RangeQuery[A, Unbounded.type, Unbounded.type] =
+  ): RangeQuery[S, A, Unbounded.type, Unbounded.type] =
     RangeQuery.empty(field.toString ++ multiField.map("." ++ _).getOrElse(""))
```
Comment (drmarjanovic) [inline modules/library/src/main/scala/zio/elasticsearch/ElasticQuery.scala:122]: Should we name params here as well?

File: modules/library/src/main/scala/zio/elasticsearch/ElasticQuery.scala [hunk 6]
```diff
@@ -70,97 +84,129 @@ object ElasticQuery {
...
       boost = None,
       caseInsensitive = None
     )
 
-  def term[A: ElasticPrimitive](field: String, value: A): ElasticQuery[Term[A]] =
+  def term[A: ElasticPrimitive](field: String, value: A): ElasticQuery[Any, Term[A]] =
     TermQuery(field = field, value = value, boost = None, caseInsensitive = None)
 
-  def wildcard(field: String, value: String): ElasticQuery[Wildcard] =
+  def wildcard(field: String, value: String): ElasticQuery[Any, Wildcard] =
```
Comment (drmarjanovic) [inline modules/library/src/main/scala/zio/elasticsearch/ElasticQuery.scala:147]: I guess we need `wildcard` with `Field` as well.

**Review by drmarjanovic — APPROVED** [review]

**Pull request merged**